### PR TITLE
network: DHCP option- use correct byteorder

### PR DESCRIPTION
**Pull request by ssahani**
Closes https://github.com/systemd/systemd/issues/18495

### CODE REVIEW

**Comment by yuwata** [comment]
LGTM.

File: src/network/networkd-dhcp-common.c [hunk 1]
```diff
@@ -683,25 +683,31 @@ int config_parse_dhcp_send_option(
...
+                uint32_t k;
+
+                r = safe_atou32(p, &k);
                 if (r < 0) {
                         log_syntax(unit, LOG_WARNING, filename, line, r,
                                    "Failed to parse DHCP uint32 data, ignoring assignment: %s", p);
                         return 0;
                 }
 
+                uint16_data = htobe32(k);
```
Comment (poettering) [inline src/network/networkd-dhcp-common.c]: uint32_data

Comment (ssahani) [reply src/network/networkd-dhcp-common.c]: oops. Fixed.

**Comment by poettering** [comment]
lgtm. thanks!

**Comment by poettering** [comment]
CI failure appears unrelated

**Pull request merged**